### PR TITLE
Fixed Code Quality Issues

### PR DESCRIPTION
**Pull request by HarshCasper**
### Description
Hi 👋 

I ran the DeepSource static analyzer on the forked copy of this repo and found some interesting code quality [issues](https://deepsource.io/gh/HarshCasper/jina/). This PR fixes a few of them.

### Summary of changes

- Use `callable()` to check if the object is calllable
- Fix dangerous default argument
- Iterate dictionary directly
- Remove redundant `None` default

### CODE REVIEW

File: jina/peapods/pods/__init__.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def _parse_args(self, args: Namespace) -> Dict[str, Optional[Union[List[Namespac
             if getattr(args, 'uses_after', None):
                 self.is_tail_router = True
                 peas_args['tail'] = _copy_to_tail_args(args)
-            peas_args['peas'] = _set_peas_args(args, peas_args.get('head', None), peas_args.get('tail', None))
+            peas_args['peas'] = _set_peas_args(args, peas_args.get('head'), peas_args.get('tail'))
```
Comment (JoanFM) [inline jina/peapods/pods/__init__.py:116]: why is this dangerous? because `_set_peas_args` do not handle properly `None` ?

**Comment by JoanFM** [comment]
Bad commit message, please refer to:
- [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
- [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)

**Comment by JoanFM** [comment]
Hey @HarshCasper , 

thank you very much for your contribution, but unfortunately your commit does not match our guidelines and we need it to be fixed before proceeding with this PR

File: .deepsource.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+version = 1
```
Comment (hanxiao) [inline .deepsource.toml:1]: this file, unless used in our CICD, should not be part of the PR

**Comment by HarshCasper** [comment]
> Hey @HarshCasper ,
> 
> thank you very much for your contribution, but unfortunately your commit does not match our guidelines and we need it to be fixed before proceeding with this PR

Sure. Is it possible to have the PR reviewed first before I reset the Commits to make them adhere to the Commit guidelines? If not, I will make the commit message changes first.

File: .deepsource.toml (hunk 2 above)
Comment (HarshCasper) [reply .deepsource.toml:1]: DeepSource uses this `.deepsource.toml` file to continuously analyze every commit and pull request for code quality checks. I have linked the report of the DeepSource analysis above in my PR. There are plenty of other issues in relation to Bug Discovery, Performance and Anti-Patterns which you would be interested to take a look at.

File: jina/peapods/pods/__init__.py (hunk 1 above)
Comment (HarshCasper) [reply jina/peapods/pods/__init__.py:116]: Yes. It is unnecessary to provide None as the default value when the key is not present as it implicitly returns None.

**Comment by JoanFM** [comment]
> > Hey @HarshCasper ,
> > thank you very much for your contribution, but unfortunately your commit does not match our guidelines and we need it to be fixed before proceeding with this PR
> 
> Sure. Is it possible to have the PR reviewed first before I reset the Commits to make them adhere to the Commit guidelines? If not, I will make the commit message changes first.

It is better to update the commit since is only do an ammend and we can see if Tests fail

**Comment by JoanFM** [comment]
Hey @HarshCasper ,

would you mind resolving the conflicts and the commit message?

**Comment by HarshCasper** [comment]
> Hey @HarshCasper ,
> 
> would you mind resolving the conflicts and the commit message?

Hi @JoanFM 

I will resolve the same by tomorrow. I had some University Exams and was caught up in that. Apologies for the delay :)
